### PR TITLE
Migrate to operator-sdk 1.2

### PR DESCRIPTION
**Pull request by DTMad**

### CODE REVIEW

File: controllers/utils/utils.go [hunk 1]
```diff
@@ -295,3 +296,23 @@ func GetField(values map[string]string, key, defaultValue string) string {
 	}
 	return defaultValue
 }
+
+func GetPod(ctx context.Context, client client.Client, ns string) (*corev1.Pod, error) {
```
Comment (lrgar) [inline controllers/utils/utils.go:300]: I'd be careful regarding (I presume) copy/pasting code from third-party libraries and license.

Just do the client.Get(... pod) command in GetDeployment. The rest here is not relevant for us.

File: hack/boilerplate.go.txt [hunk 2]
```diff
@@ -1,6 +1,4 @@
 /*
-Copyright YEAR Dynatrace LLC.
-
```
Comment (lrgar) [inline hack/boilerplate.go.txt:3]: Pretty certain you want to leave this.

File: main.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2020 Dynatrace LLC.
+
```
Comment (lrgar) [inline main.go:2]: Put it back.

File: operator.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		nodes.Add,
+	} {
+		if err := f(mgr, ns); err != nil {
+			return nil, err
+		}
+	}
+
+	go func() {
+		log.Info("serving operator probe endpoint on :10080/healthz")
+		err := http.ListenAndServe(":10080", http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
```
Comment (lrgar) [inline operator.go:63]: Hmm, manager.Manager already has functions for health checks, e.g., [`AddHealthzCheck` and `AddReadyzCheck`](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.7.0/pkg/manager#Manager), but haven't checked how they work.

File: webhook_server.go [hunk 5]
```diff
@@ -21,18 +21,23 @@ import (
 	"path"
 	"time"
 
-	"github.com/Dynatrace/dynatrace-oneagent-operator/pkg/apis"
-	"github.com/Dynatrace/dynatrace-oneagent-operator/pkg/webhook/server"
+	"github.com/Dynatrace/dynatrace-oneagent-operator/webhook/server"
+	"github.com/prometheus/common/log"
```
Comment (lrgar) [inline webhook_server.go:25]: Wrong log package

File: webhook_bootstrapper.go [hunk 6]
```diff
@@ -17,29 +17,30 @@ limitations under the License.
 package main
 
 import (
-	"github.com/Dynatrace/dynatrace-oneagent-operator/pkg/apis"
-	"github.com/Dynatrace/dynatrace-oneagent-operator/pkg/webhook/bootstrapper"
+	"github.com/Dynatrace/dynatrace-oneagent-operator/webhook/bootstrapper"
+	"github.com/prometheus/common/log"
```
Comment (lrgar) [inline webhook_bootstrapper.go:21]: Wrong log import. Don't you want an instance instead?

File: webhook/bootstrapper/controller.go [hunk 7]
```diff
@@ -86,6 +82,14 @@ func add(mgr manager.Manager, r *ReconcileWebhook) error {
 		}
 	}()
 
+	if err = mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		r.logger.Error(err, "could not start health endpoint for operator")
+	}
+
+	if err = mgr.AddReadyzCheck("healthz", healthz.Ping); err != nil {
```
Comment (lrgar) [inline webhook/bootstrapper/controller.go:89]: Did you mean `"readyz"`?

Comment (DTMad) [reply webhook/bootstrapper/controller.go]: Fixed

**Comment by DTMad** [comment]
This won't be merged before the release of v0.9.4

**Review by lrgar — APPROVED** [review]

**Pull request merged**